app/src/main/java/de/xikolo/network/jobs/base/NetworkStateLiveData.kt

```diff
@@ -17,22 +18,34 @@ class NetworkStateLiveData : LiveData<NetworkState>() {
         state(NetworkCode.ERROR, userRequest)
     }
 
-    fun state(code: NetworkCode, userRequest: Boolean) {
+    fun deprecated(deprecationDate: Date, userRequest: Boolean) {
+        state(NetworkCode.API_VERSION_DEPRECATED, userRequest, deprecationDate)
+    }
+
+    fun apiVersionExpired(userRequest: Boolean) {
+        state(NetworkCode.API_VERSION_EXPIRED, userRequest)
+    }
+
+    fun maintenance(userRequest: Boolean) {
+        state(NetworkCode.MAINTENANCE, userRequest)
```

Comment:
Same here. How about a sub-class of `NetworkStateLiveData`?

Reply:
I tried to subclass `NetworkStateLiveData` including `NetworkState` and `NetworkCode`. The problem with that is, that the `enum class NetworkCode` cannot be inherited from, effectively making subclassing impossible.

Reply:
One way of solving this would be to replace the `enum` fields with Int constants but this would reqire changes throughout the entire codebase.

Reply:
You don't need to subclass `NetworkCode`. Just move it to a separate file. It is fine if it contains values that are not used by the base class `NetworkStateLiveData` and only by a subclass of it.